mlflow/store/artifact/databricks_logged_model_artifact_repo.py

```diff
@@ -0,0 +1,156 @@
+from databricks.sdk.config import Config
+from databricks.sdk.errors.platform import NotFound
+from databricks.sdk.service.files import FilesAPI
+
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.artifact.databricks_artifact_repo import DatabricksArtifactRepository
+
+_logger = logging.getLogger(__name__)
+
+
+class DatabricksLoggedModelArtifactRepository(ArtifactRepository):
+    # Matches URIs of the form:
+    # databricks/mlflow-tracking/<experiment_id>/logged_models/<model_id>/<relative_path>
+    _URI_REGEX = re.compile(
+        r"databricks/mlflow-tracking/(?P<experiment_id>[^/]+)/logged_models/(?P<model_id>[^/]+)(?P<relative_path>/.*)?$"
+    )
+
+    @staticmethod
+    def is_logged_model_uri(artifact_uri):
+        return bool(DatabricksLoggedModelArtifactRepository._URI_REGEX.search(artifact_uri))
+
+    def __init__(self, artifact_uri: str) -> None:
+        super().__init__(artifact_uri)
+        m = self._URI_REGEX.search(artifact_uri)
+        if not m:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid artifact URI: {artifact_uri}. Expected URI of the form "
+                f"databricks/mlflow-tracking/<EXP_ID>/logged_models/<MODEL_ID>"
+            )
+        experiment_id = m.group("experiment_id")
+        model_id = m.group("model_id")
+        relative_path = m.group("relative_path") or ""
+        self.root_path = f"/Mlflow/Artifacts/{experiment_id}/LoggedModels/{model_id}{relative_path}"
+        self.databricks_artifact_repo = DatabricksArtifactRepository(artifact_uri)
+        self.wc = WorkspaceClient(config=Config(enable_experimental_files_api_client=True))
+
+    @property
+    def files_api(self) -> FilesAPI:
+        return self.wc.files
+
+    def _is_dir(self, path: str) -> bool:
+        try:
+            self.files_api.get_directory_metadata(path)
+        except NotFound:
+            return False
+        return True
+
+    def path(self, artifact_path: str) -> str:
+        """
+        Construct the full path to the artifact, given the artifact's relative path.
+        """
+        return f"{self.root_path}/{artifact_path}" if artifact_path else self.root_path
+
+    def _log_artifact(self, local_file: str, artifact_path: Optional[str] = None) -> None:
+        with open(local_file, "rb") as f:
+            self.files_api.upload(self.path(artifact_path), f, overwrite=True)
+
+    def log_artifact(self, local_file: str, artifact_path: Optional[str] = None) -> None:
+        try:
+            self._log_artifact(local_file, artifact_path)
+        except Exception:
+            _logger.info(
+                (
+                    "Failed to log artifact using Databricks SDK, falling back to "
+                    "DatabricksArtifactRepository"
+                ),
+                exc_info=True,
+            )
```

Comment:
There's a redundant call to `self.databricks_artifact_repo.log_artifact()` here. The method is called and then returned on the next line, which will cause the operation to be performed twice.